web-ui/src/streaming/import/random-data/index.tsx

```diff
@@ -0,0 +1,194 @@
+// Displays a form with random settings to configure what values are generated
+// for a given field in a table.
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Autocomplete, FormHelperText, Grid, TextField } from '@mui/material'
+import { Controller, useWatch } from 'react-hook-form'
+import dayjs, { Dayjs } from 'dayjs'
+
+import { useCustomForm } from '../hooks/useCustomForm'
+import { Field } from 'src/types/manager'
+import { getValueParser } from 'src/types/ddl'
+import { IRngGenMethod, columnTypeToRngOptions, getRngMethodByName } from './generators'
+import { ExampleDisplay } from './ExampleDisplay'
+import { StoredFieldSettings } from '../RngSettingsDialog'
+import { StoreSettingsFn } from '../ImportToolbar'
+
+// The names of potential form field parameters.
+export enum FieldNames {
+  BOOL_CONST = 'bool_const',
+  TRUE_PCT = 'true_pct',
+  VALUE = 'value',
+  MIN = 'min',
+  MAX = 'max',
+  MU = 'mu',
+  SIGMA = 'sigma',
+  LAMBDA = 'lambda',
+  ALPHA = 'alpha',
+  BETA = 'beta',
+  TIME = 'time',
+  TIME2 = 'time2',
+  DATE_RANGE = 'date_range'
+}
+
+export const RngFieldSettings = (props: {
+  field: Field
+  fieldSettings: StoredFieldSettings | undefined
+  setSettings: StoreSettingsFn
+}) => {
+  const { field, fieldSettings, setSettings } = props
+
+  const defaultValues: Partial<Record<FieldNames, any>> = useMemo(() => {
+    return {
+      bool_const: false,
+      true_pct: '0.5',
```

Comment:
strings?

Reply:
because it's coming from a HTML form the types are always strings... :(